### DISCUSSION TITLE
Difference between GLX and EGL Version

### DISCUSSION THREAD

**Discussion by pvmilk** [selkies-project/docker-nvidia-glx-desktop#21] [Q&A] [answered]
Hello, 
Thanks for a nice work and a push in nvidia-docker community. 

I have some question regards this version of your docker and the `docker-nvidia-egl-desktop` version. 
For the EGL version, I can understand that it would required `/dev/dri/*` to be able to access the GPU, but
1. what do you mean by GLX and Vulkan? and any reference explaining the big picture of them and the EGL?
2. apart from the driver version, what device does it required? (I also would like to have it in the container).

Thanks

**Comment by pvmilk** [comment]
For the first question, I think [this article](https://utcc.utoronto.ca/~cks/space/blog/linux/EGLAndGLXAndOpenGL) probably answered it.

**Comment by ehfd** [comment, marked as answer]
Hi! Please refer to https://www.researchgate.net/publication/353658640_GPU_Remote_Desktop_Container_Project for the differences. I generally recommend docker-nvidia-glx-desktop for most cases including Kubernetes, and docker-nvidia-egl-desktop when you need software fallback when the GPU is not available (or even use AMD or Intel GPUs!), and in special situations including ARM CPUs. All you need is the NVIDIA container runtime set up and to use `--gpu 1` or an equivalent argument, because NVIDIA uses a different method to the `/dev/dri/cardX` device.
